### ISSUE TITLE
ssl_client2 crashes with early_data=1

### ISSUE THREAD

**Issue by zhihan** [hannestschofenig/mbedtls#250]
```
./ssl_client2 server_name=localhost server_port=1234 auth_mode=none force_ciphersuite=TLS_AES_256_GCM_SHA384 key_share_named_groups=secp384r1 reconnect=1 early_data=1 debug_level=3
```

./ssl_client2 server_name=localhost server_port=1234 auth_mode=none force_ciphersuite=TLS_AES_256_GCM_SHA384 reconnect=1 early_data=1 debug_level=3

  . Seeding the random number generator... ok
  . Loading the CA root certificate ... ok (0 skipped)
  . Loading the client cert. and key... ok (key type: EC)
  . Connecting to tcp/localhost/1234... ok
  . Setting up the SSL/TLS structure...ssl_msg.c:0073: |3| set_timer to 0 ms
 ok
  . Performing the SSL/TLS handshake...ssl_tls.c:6784: |2| => handshake
ssl_tls13_client.c:3792: |2| client state: 0
ssl_tls13_client.c:3792: |2| client state: 1
ssl_tls13_client.c:1341: |2| => write client hello
ssl_tls13_client.c:1608: |3| dumping 'client hello, random bytes' (32 bytes)
ssl_tls13_client.c:1608: |3| 0000:  ba ae 43 37 7c 18 d8 7d 07 81 4c 72 f9 1b b8 ad  ..C7|..}..Lr....
ssl_tls13_client.c:1608: |3| 0010:  91 24 4d 17 7e 99 04 60 2f 37 a9 06 54 e7 15 88  .$M.~..`/7..T...
ssl_tls13_client.c:1639: |3| session id len.: 32
ssl_tls13_client.c:1640: |3| dumping 'session id' (32 bytes)
ssl_tls13_client.c:1640: |3| 0000:  81 e1 0e ad 94 e8 2b 3f 94 95 45 9a c8 f7 19 c4  ......+?..E.....
ssl_tls13_client.c:1640: |3| 0010:  a3 5c ab b1 2b c1 01 fc 04 2d a8 bc 51 d9 21 41  .\..+....-..Q.!A
ssl_tls13_client.c:1686: |3| client hello, add ciphersuite: 1302, TLS_AES_256_GCM_SHA384
ssl_tls13_client.c:1711: |3| client hello, got 1 ciphersuites
ssl_tls13_client.c:0512: |3| client hello, adding supported version extension
ssl_tls13_client.c:0536: |3| supported version: [3:4]
ssl_tls13_client.c:1009: |3| no cookie to send; skip extension
ssl_tls13_generic.c:3122: |3| client hello, adding early_data extension
ssl_tls13_client.c:0447: |3| client hello, adding server name extension: localhost
ssl_tls13_client.c:0696: |3| client hello, adding psk_key_exchange_modes extension
ssl_tls13_client.c:1103: |3| client hello, adding supported_groups extension
ssl_tls13_client.c:1138: |3| dumping 'Supported groups extension' (24 bytes)
ssl_tls13_client.c:1138: |3| 0000:  00 16 00 19 00 1c 00 18 00 1b 00 17 00 16 00 1a  ................
ssl_tls13_client.c:1138: |3| 0010:  00 15 00 14 00 13 00 12                          ........
ssl_tls13_generic.c:0351: |3| adding signature_algorithms extension
ssl_tls13_generic.c:0381: |3| signature scheme [403]
ssl_tls13_generic.c:0381: |3| signature scheme [503]
ssl_tls13_client.c:1216: |3| client hello, adding key share extension
ssl_tls13_client.c:1268: |2| ECDHE curve: secp256r1
ssl_tls13_client.c:1290: |3| value of 'ECDHE: Q (X)' (255 bits) is:
ssl_tls13_client.c:1290: |3|  73 cd c0 ed 52 93 42 49 2d 77 ad 91 ac 3b d1 b7
ssl_tls13_client.c:1290: |3|  8c c8 b0 76 83 8d eb 20 2f 2f 0e ac 0c 7f 1f a8
ssl_tls13_client.c:1290: |3| value of 'ECDHE: Q (Y)' (256 bits) is:
ssl_tls13_client.c:1290: |3|  e8 78 98 c4 70 28 56 54 59 41 49 7d fb c6 53 b1
ssl_tls13_client.c:1290: |3|  62 1c a6 74 f4 b6 a4 cd 25 4b e8 70 e0 29 45 3a
ssl_tls13_client.c:0793: |3| client hello, skip pre_shared_key extensions
ssl_tls13_client.c:1849: |3| client hello, total extension length: 149
ssl_tls13_client.c:1851: |3| dumping 'client hello extensions' (149 bytes)
ssl_tls13_client.c:1851: |3| 0000:  00 00 00 2b 00 03 02 03 04 00 2a 00 00 00 00 00  ...+......*.....
ssl_tls13_client.c:1851: |3| 0010:  0e 00 0c 00 00 09 6c 6f 63 61 6c 68 6f 73 74 00  ......localhost.
ssl_tls13_client.c:1851: |3| 0020:  2d 00 03 02 00 01 00 0a 00 18 00 16 00 19 00 1c  -...............
ssl_tls13_client.c:1851: |3| 0030:  00 18 00 1b 00 17 00 16 00 1a 00 15 00 14 00 13  ................
ssl_tls13_client.c:1851: |3| 0040:  00 12 00 0d 00 06 00 04 04 03 05 03 00 33 00 47  .............3.G
ssl_tls13_client.c:1851: |3| 0050:  00 45 00 17 00 41 04 73 cd c0 ed 52 93 42 49 2d  .E...A.s...R.BI-
ssl_tls13_client.c:1851: |3| 0060:  77 ad 91 ac 3b d1 b7 8c c8 b0 76 83 8d eb 20 2f  w...;.....v... /
ssl_tls13_client.c:1851: |3| 0070:  2f 0e ac 0c 7f 1f a8 e8 78 98 c4 70 28 56 54 59  /.......x..p(VTY
ssl_tls13_client.c:1851: |3| 0080:  41 49 7d fb c6 53 b1 62 1c a6 74 f4 b6 a4 cd 25  AI}..S.b..t....%
ssl_tls13_client.c:1851: |3| 0090:  4b e8 70 e0 29                                   K.p.)
ssl_tls13_client.c:0793: |3| client hello, skip pre_shared_key extensions
ssl_tls13_client.c:1476: |2| <= write client hello
ssl_tls13_client.c:3792: |2| client state: 24
ssl_tls13_generic.c:0175: |2| => write change cipher spec
ssl_tls13_generic.c:0214: |2| <= write change cipher spec
ssl_tls13_client.c:3792: |2| client state: 27
ssl_tls13_client.c:0132: |2| => write early data
ssl_tls13_keys.c:0558: |2| => mbedtls_ssl_generate_early_data_keys
zsh: segmentation fault  ./ssl_client2 server_name=localhost server_port=1234 auth_mode=none

**Comment by lhuang04** [comment]
Could be the same issue found in #118

**Comment by zhihan** [comment]
Lin, thanks for the help. That PR fixes the crash.

**Comment by hanno-becker** [comment]
This has been fixed by #118.
